### DISCUSSION TITLE
Storing Component into Zustand store

### DISCUSSION THREAD

**Discussion by BenSeward** [pmndrs/zustand#1911] [Q&A]
Hi All, I'd like to start off by saying thank you for making such a brilliant state management package. I've really been enjoying using it!

That being said, I have a problem. I have an error modal component that I want to be able to render anything that is put within the store. 

The idea is that I can dispatch an event, and pass a component into the Zustand store and it then feeds into the error component. Currently it works until I refresh the app, then the following error within Modal.tsx:

`Error: Objects are not valid as a React child (found: object with keys {key, ref, props, _owner, _store}). If you meant to render a collection of children, use an array instead.`

**WorkoutHistoryCard.tsx**
```
import {useStore} from '../../store';

export const WorkoutHistoryCard = ({workoutHistory}: WorkoutItemProps) => {
  const {dispatchModalContent} = useStore();

  return (
    <Pressable
      onPress={() =>
        dispatchModalContent({
          action: ModalContentActions.UPDATE,
          args: {title: workoutHistory.workoutTitle, content: <ModalContent />},
        })
      }
      style={styles.container}>
      ...
    </Pressable>
  );
};

const ModalContent = () => {
  return (
    <View style={styles.modal}>
      <WorkoutSummary workoutId={"workoutId"} />
    </View>
  );
};
```

**Modal.tsx**
```
import {useStore} from '../../store';

export const Modal = () => {
  const {modalContent} = useStore();

  if (!modalContent) {
    return null;
  }

  return (
    <View style={styles.modalContainer}>
        {modalContent.content}
    </View>
  );
};
```

Any help would be greatly appreciated! Thank you

**Comment by dbritto-dev** [comment]
@BenSeward btw, did you tried passing the function component itself? `content: ModalContent` and then you can use it like this `<modalContent.content />`

**Reply by BenSeward** [reply]
I did yeah, but unfortunately I think the problem is actually due to React re-hydrating and nothing to do with Zustand. Going to close this issue.
